src/main/java/net/mcreator/themes/ThemeLoader.java

```diff
@@ -0,0 +1,142 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2021, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.themes;
+
+import com.google.gson.Gson;
+import net.mcreator.io.FileIO;
+import net.mcreator.plugin.PluginLoader;
+import net.mcreator.preferences.PreferencesManager;
+import net.mcreator.ui.init.UIRES;\
+import net.mcreator.util.image.ImageUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.swing.*;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class ThemeLoader {
+	private static final Logger LOG = LogManager.getLogger("Theme Loader");
+
+	private static final LinkedHashSet<Theme> THEMES = new LinkedHashSet<>();
+	public static Theme CURRENT_THEME;
+
+	public static void initUIThemes() {
+		LOG.debug("Loading UI themes");
+
+		final Gson gson = new Gson();
+
+		// Load themes
+		Set<String> files = PluginLoader.INSTANCE.getResources("themes", Pattern.compile("theme.json"));
+		for (String file : files) {
+			Theme theme = gson.fromJson(FileIO.readResourceToString(PluginLoader.INSTANCE, file), Theme.class);
+			// The ID will be used to get images from this theme if the user select it.
+			theme.setId(new File(file).getParentFile().getName());
+
+			if (theme.getColorScheme() != null) {
+				// Set the value for the Blockly Panel file
+				// If the file does not exist, we use the default file (from the default_dark them)
+				if (PluginLoader.INSTANCE.getResource(
+						"themes/" + theme.getID() + "/colors/blockly_" + theme.getColorScheme().getID() + ".css") == null) {

```

Comment:
This check is no longer correct. Fix and test please.

Also I would move XML and CSS files to eg. style folder, not colors as they can do more than just colors